edx_api/enrollments/models.py

```diff
@@ -0,0 +1,182 @@
+        """
+        Returns a generator of all the courses the user has enrolled in
+        """
+        for enrollment_json in self.payload:
+            yield Enrollment(enrollment_json)
+
+
+@python_2_unicode_compatible
+class Enrollment(object):
+    """
+    Single enrollment object representation
+    """
+    def __init__(self, json):
+        self.json = json
+
+    def __str__(self):
+        return "<Enrollment for user {user} in course {course}>".format(
+            user=self.user,
+            course=self.course_details.course_id
+        )
+
+    @property
+    def created(self):
+        """Returns a string representation of the enrollment timestamp"""
+        return self.json.get('created')
+
+    @property
+    def mode(self):
+        """Returns the enrollment mode of the user in this course"""
+        return self.json.get('mode')
+
+    @property
+    def is_active(self):
+        """Returns whether the enrollment is currently active."""
+        return self.json.get('is_active') is True
+
+    @property
+    def course_details(self):
+        """Returns a CourseDetails object"""
+        return CourseDetails(self.json.get('course_details', {}))
+
+    @property
+    def user(self):
+        """Returns the username of the user."""
+        return self.json.get('user')
+
+
+@python_2_unicode_compatible
+class CourseDetails(object):
+    """
+    Course enrollment info
+    """
+    def __init__(self, json):
+        self.json = json
+
+    def __str__(self):
+        return "<Enrollment details for course {}>".format(self.course_id)
+
+    @property
+    def course_id(self):
+        """Returns the unique identifier for the course."""
+        return self.json.get('course_id')
+
+    @property
+    def course_start(self):
+        """
+        Returns the date and time when the course opens.
+        If None, the course opens immediately when it is created.
+        """
+        return self.json.get('course_start')
+
+    @property
+    def course_end(self):
+        """
+        Returns the date and time when the course closes.
+        If None, the course never ends.
+        """
+        return self.json.get('course_end')
+
+    @property
+    def enrollment_start(self):
+        """
+        Returns the date and time when users can begin enrolling in the course.
+        If None, enrollment opens immediately when the course is created.
+        """
+        return self.json.get('enrollment_start')
+
+    @property
+    def enrollment_end(self):
+        """
+        Returns the date and time after which users cannot enroll for the course.
+        If None, the enrollment period never ends.
+        """
+        return self.json.get('enrollment_end')
+
+    @property
+    def invite_only(self):
+        """
+        Returns a boolean indicating whether students
+        must be invited to enroll in the course.
+        """
+        return self.json.get('invite_only') is True
+
+    @property
+    def course_modes(self):
```

Comment:
My vote is to delete this method and the model below. It doesn't strike me as particularly useful, but I don't feel very strongly about this.